File: frontend/replace_theme.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const directory = path.join(__dirname, 'src');

const pattern = /\b(bg|text|border|ring|divide)-blue-(\d+)\b/g;
const patternHover = /\b(hover:bg|hover:text|hover:border|focus:ring|focus:border|active:bg|group-hover:text)-blue-(\d+)\b/g;

let countFilesModified = 0;

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(function(file) {
    file = dir + '/' + file;
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) { 
      results = results.concat(walk(file));
    } else { 
      results.push(file);
    }
  });
  return results;
}

const files = walk(directory).filter(f => f.match(/\.(tsx|ts|jsx|js)$/));

files.forEach(filepath => {
  const content = fs.readFileSync(filepath, 'utf-8');
  let newContent = content.replace(pattern, '$1-emerald-$2');
  newContent = newContent.replace(patternHover, '$1-emerald-$2');
  newContent = newContent.replace(/badge-blue/g, 'badge-sage');

  if (newContent !== content) {
    fs.writeFileSync(filepath, newContent, 'utf-8');
    countFilesModified++;
  }
});

console.log(`Modified ${countFilesModified} files.`);

File: frontend/replace_theme.py
import os
import re

def replace_in_files(directory):
    # Matches words like bg-blue-500, text-blue-600, border-blue-400, ring-blue-500, etc.
    # Also hover:bg-blue-600, etc.
    pattern = re.compile(r'\b(bg|text|border|ring|divide)-blue-(\d+)\b')
    pattern_hover = re.compile(r'\b(hover:bg|hover:text|hover:border|focus:ring|focus:border|active:bg)-blue-(\d+)\b')
    
    # We will also change badge-blue to badge-sage or keep badge-blue but defined in index.css as sage.
    # Actually, we should just let the regex catch the primary tailwind classes.

    count_files_modified = 0
    count_replacements = 0

    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.tsx', '.ts', '.jsx', '.js')):
                filepath = os.path.join(root, file)
                
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()

                # Perform replacement
                new_content = pattern.sub(r'\1-emerald-\2', content)
                new_content = pattern_hover.sub(r'\1-emerald-\2', new_content)
                
                # Also replace arbitrary blue texts where it matters like text-blue
                new_content = new_content.replace('badge-blue', 'badge-sage')
                new_content = new_content.replace('bg-blue-50', 'bg-emerald-50')
                new_content = new_content.replace('text-blue-600', 'text-emerald-600')

                # Some specific overrides to amber (gold) if there are any specific highlights we want, but emerald is fine for now globally.
                
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    count_files_modified += 1
                    # count approx diff
                    count_replacements += len(content) - len(new_content) # just an indicator

    print(f"Modified {count_files_modified} files.")

if __name__ == '__main__':
    src_dir = os.path.join(os.path.dirname(__file__), 'src')
    replace_in_files(src_dir)
